### ISSUE TITLE
Eslint falsely reports matching configured ignore pattern for implicit ignore pattern

### ISSUE THREAD

**Issue by snhardin** [eslint/eslint#12873]
**Tell us about your environment**

* **ESLint Version:** 7.0.0-alpha.0
* **Node Version:** v12.14.1
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "extends": "airbnb-base"
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I was told that an additional issue needed to be opened for code changes. #12348 has been scoped to be documentation changes only. That issue has the original instructions to reproduce the issue, although I have been able to successfully reproduce the issue myself.

If I have a file located in a dot directory (for example, `client/.storybook/config.js`), I can lint it by telling eslint to analyze the directory it lives in (`npx eslint client/.storybook`). However, if I try to lint the file directory, I am incorrectly told that the file has been ignored because of a matching ignore pattern. I have no ignore patterns defined, so this is incorrect. The file was actually ignored due to an implicit ignore rule.

Source: N/A

```bash
npx eslint client/.storybook/config.js
```

**What did you expect to happen?**

Eslint should indicate that the file was ignored by default.

**What actually happened? Please include the actual, raw output from ESLint.**

Eslint indicated that the file was ignored because of a configured ignore pattern. I have no ignore patterns configured.

**Are you willing to submit a pull request to fix this bug?**

Yes

**Comment by kaicataldo** [comment]
Thanks for reporting this. Given the fact that we show a more helpful error message for ignored dotfiles, I agree that this seems like a bug. It looks to me like we should update [this logic](https://github.com/eslint/eslint/blob/master/lib/cli-engine/cli-engine.js#L279-L291) to check if the file is located in a dotfolder and report an appropriate error message for that.

Maybe something like `"File is in an ignored directory by default.  Use a negated ignore pattern (like \"--ignore-pattern '!<relative/path/to/directory>'\") to override."`?
